src/utils/http.ts

```diff
@@ -0,0 +1,131 @@
+enum METHODS {
+    GET = "GET",
+    PUT = "PUT",
+    POST = "POST",
+    DELETE = "DELETE"
+}
+
+// unknown может спровоцировать ошибку
+function queryStringify(data: Record<string, unknown>) {
+	const entries = Object.entries(data).map(
+		([key, val]) => (`${key}=${val}`)
+	);
+	return `?${entries.join("&")}`;
+}
+
+export default class HTTP {
+
+	private _baseURL: string;
+
+	constructor(baseURL: string) {
+		this._baseURL = baseURL;
+	}
+
+	private _getFullUrl = (endpoint: string) => this._baseURL + endpoint;
+
+	async get<T>(
+		endpoint: string,
+		options?: {
+			data: any,
+			timeout?: number
+		}): Promise<T> {
+			const queryUrl = options && options.data ? `${endpoint}${queryStringify(options.data)}` : endpoint;
+			return this.request(
+				queryUrl,
+				{...options, method: METHODS.GET}
+			);
+	};
+
+	async put<T>(
+		endpoint: string,
+		options?: {
+			data: any,
+			timeout?: number
+		}): Promise<T> {
+			return this.request(
+				endpoint,
+				{...options, method: METHODS.PUT}
+			);
+	}
+
+	async post<T>(
+		endpoint: string,
+		options?: {
+			data: any,
+			timeout?: number
+		}): Promise<T> {
+			return this.request(
+				endpoint,
+				{...options, method: METHODS.POST}
+			);
+	}
+
+	async delete<T>(
+		endpoint: string,
+		options?: {
+			data: any,
+			timeout?: number
+		}): Promise<T> {
+			return this.request(
+				endpoint,
+				{...options, method: METHODS.DELETE}
+			);
+	}
+
+	async request<T>(
+		endpoint: string,
+		options: {
+			method: METHODS,
+			data?: any,
```

Comment:
Если мы в коде ничего не знает про data и ничего из data не читаем, то лучше использовать unknown. Это как any, только безопаснее
```suggestion
			data?: unknown,
```